### DISCUSSION TITLE
Client ID always includes corresponding projects, not usable with OAuth

### DISCUSSION THREAD

**Discussion by TobiasGrether** [zitadel/zitadel#5934] [Ideas]
The ClientID for applications is set to be a number with an @ and the project name after it. This might be fine for some implementation, but services like Auth0 actually enforce a regex on ClientIDs that doesnt allow for special characters like @.
This is an issue because it essentially makes Zitadel unusable with Auth0.

Ideally, you should be able to toggle the behaviour of adding the @{projectName} on or off or allow the user to change the ClientID manually (error-prone though).

**Comment by fforootd** [comment]
You use ZITADEL as IdP to Auth0 where this problem happens, right?

@livio-a what do you think if we would align the clientID with the rest of ZITADELs IDs. I.e ditch the @XYZ

@TobiasGrether The suffix was primarily intended as means for humans to quickly see to what project a client belongs to.

**Reply by mffap** [reply]
See also https://discord.com/channels/927474939156643850/927866013545025566/1111596650121670679

The standard does not put a restriction on the characters. This seems an issue with Auth0 actually (and not generally OAuth). See: 
https://www.rfc-editor.org/rfc/rfc6749#appendix-A.1
